pinot-filesystem/src/main/java/com/linkedin/pinot/filesystem/PinotFS.java

```diff
@@ -88,10 +88,11 @@
    * Throws exception if this abstract pathname is not valid, or if
    * an I/O error occurs.
    * @param fileUri location of file
+   * @param recursive if we want to list files recursively
    * @return an array of strings that contains file paths
    * @throws IOException see specific implementation
    */
-  public abstract String[] listFiles(URI fileUri) throws IOException;
+  public abstract String[] listFiles(URI fileUri, boolean recursive) throws IOException;
```

Comment:
Since you are changing the interface, consider adding a filter as an argument. (e.g. to return only files or only directories or both). checking whether a returned object is a file or dir after the fact could prove to be expensive if remote file system.

Reply:
I like this idea; I will think about how to design this/add this in another one.